src/machine_learning/perceptron/single_layer_perceptron.rs

```diff
@@ -0,0 +1,103 @@
+// Perceptron model implementation.
+// A perceptron is a basic unit of a neural network, capable of binary classification.
+// It takes inputs, applies weights, and passes the result through an activation function to produce an output.
+
+use crate::machine_learning::perceptron::ActivationFunction;
+use rand::prelude::*;
+
+pub struct Perceptron {
+    weights: Vec<f64>,                 // Weights assigned to each input feature
+    learning_rate: f64,                // Learning rate for weight adjustment during training
```

Comment:
Why do you have this field here? A `Perceptron` should not know about its `learning_rate`. This is used only while learning. Teaching and using a `Perceptron` are two different things and should not be mixed.